precompiles/EcPairing.yul

```diff
@@ -384,19 +411,24 @@ object "EcPairing" {
 
             /// @notice Checks if a G2 point in affine coordinates belongs to the twisted curve.
             /// @dev The coordinates are encoded in Montgomery form.
-            /// @dev In Affine coordinates the point belongs to the curve if it satisfies the equation: y^3 = x^2 + 3/(9+u).
-            /// @dev The point is assumed not to be the point at infinity.
+            /// @dev in Affine coordinates the point belongs to the curve if it satisfies the equation: y^3 = x^2 + 3/(9+u).
             /// @dev See https://hackmd.io/@jpw/bn254#Twists for further details.
             /// @param x0, x1 The x coordinate to check.
             /// @param y0, y1 The y coordinate to check.
             /// @return ret True if the point is in the curve, false otherwise.
-			function g2AffinePointIsOnCurve(x0, x1, y0, y1) -> ret {
-                let a0, a1 := MONTGOMERY_TWISTED_CURVE_COEFFS()
-                let b0, b1 := fp2Mul(x0, x1, x0, x1)
-                b0, b1 := fp2Mul(b0, b1, x0, x1)
-                b0, b1 := fp2Add(b0, b1, a0, a1)
-                let c0, c1 := fp2Mul(y0, y1, y0, y1)
-                ret := and(eq(b0, c0), eq(b1, c1))
+            function g2AffinePointIsOnCurve(x0, x1, y0, y1) -> ret {
+                // FIXME: this check should be outside of the function
+                if g2AffinePointIsInfinity(x0, x1, y0, y1) {
+                    ret := 1
+                }
+                if iszero(g2AffinePointIsInfinity(x0, x1, y0, y1)) {
+                    let a0, a1 := MONTGOMERY_TWISTED_CURVE_COEFFS()
+                    let b0, b1 := fp2Mul(x0, x1, x0, x1)
+                    b0, b1 := fp2Mul(b0, b1, x0, x1)
+                    b0, b1 := fp2Add(b0, b1, a0, a1)
+                    let c0, c1 := fp2Mul(y0, y1, y0, y1)
+                    ret := and(eq(b0, c0), eq(b1, c1))
+                }
```

Comment:
This is deprecated, the infinity check is done before checking that the point is on the curve (see [here](https://github.com/lambdaclass/zksync_era_precompiles/blob/main/precompiles/EcPairing.yul#L1428-L1441)). Unless there's another reason for this function to have the check we should remove it (because of performance and correctness).

Reply:
You're right, there's no use for that check in there. Removed!